src/reducers/platform.js

```diff
@@ -19,45 +20,52 @@ import {
 	CACHE_SUCCESS
 } from '../actions/cacheActionCreators';
 
+type ApiState = {
+	inFlight: Array<string>,
+	fail?: boolean,
+	[string]: QueryResponse,
+};
+
+export const DEFAULT_API_STATE: ApiState = { inFlight: [] };
 export const DEFAULT_APP_STATE = { isFetching: false };
 
-export const responseToState = response => {
-	const { ref, ...data } = response;
-	return { [ref]: data };
-};
+export const responseToState = (response: QueryResponse): { [string]: QueryResponse } =>
+	({ [response.ref]: response });
 
-/**
+/*
  * The primary reducer for data provided by the API
- * `state.api` sub-tree
- *
- * @param {Object} state
- * @param {ReduxAction} action
- * @return {Object}
  */
-export function api(state=DEFAULT_APP_STATE, action={}) {
+export function api(state: ApiState = DEFAULT_API_STATE, action: FluxStandardAction): ApiState {
 	switch (action.type) {
-	case API_REQ:
+	case API_REQ: {
+		const requestRefs = (action.payload || []).map(({ ref }) => ref);
+		const inFlight = state.inFlight
+			.filter(ref => !requestRefs.includes[ref]) // clean out current duplicates
+			.concat(requestRefs); // add new requested refs
```

Comment:
it's more structural than technical, but to my mind an array represents the 'flow' of `ref`s a little more accurately - we are essentially representing the 'queue' network requests, ordered by time. Queue/List/Array matches that analogy. The data structure is probably also a symptom of my preference for functional over OO solutions - managing mutability of state represented by an object is a little more bug prone because `state.inFlight[ref] = true` seems intuitive, but `state.inFlight = { ...state.inFlight, [ref]: true }` is actually preferred for immutability.

Objects also bring up a question of how to correctly represent a completed `ref` - do you set it to `false` or do you `delete` the key? What are the consequences of one approach vs the other? The object approach ultimately feels like a more complex one, particularly with the nice `Array.prototype` methods that are available for immutably updating arrays.

I'm open to finding out more about what problems an object-based representation would solve, but in the consumer code I don't think it would be much simpler - `inFlight.includes(ref) ?` vs `inFlight[ref] ?`